src/hooks/portfolio/usePortfolioDeFiPositions.ts

```diff
@@ -90,11 +89,29 @@ export const usePortfolioDeFiPositions = ({
       return undefined;
     }
 
-    const allPositions: DefiPosition[] = queries
-      .filter((query) => query.isSuccess && query.data)
-      .flatMap((query) => query.data?.positions ?? []);
-
-    return { positions: allPositions };
+    const successfulQueries = queries.filter(
+      (query) => query.isSuccess && query.data,
+    );
+
+    const allPositions: DefiPosition[] = successfulQueries.flatMap(
+      (query) => query.data?.data ?? [],
+    );
+
+    // Extract dates from all queries (meta.updatedAt is ISO string, dataUpdatedAt is timestamp)
+    const dates = successfulQueries.map((query) => {
+      const metaUpdatedAt = query.data?.meta?.updatedAt;
+      return metaUpdatedAt
+        ? new Date(metaUpdatedAt)
+        : new Date(query.dataUpdatedAt);
+    });
+
+    const oldestUpdatedAt =
+      dates.length > 0 ? min(dates).toISOString() : undefined;
+
+    return {
+      data: allPositions,
+      meta: { updatedAt: oldestUpdatedAt ?? new Date().toISOString() },
+    };
```

Comment:
I think it makes sense to directly assign the fallback date to `oldestUpdatedAt` rather than do it in the return statement

```suggestion
    const oldestUpdatedAt =
      dates.length > 0 ? min(dates).toISOString() : new Date().toISOString();

    return {
      data: allPositions,
      meta: { updatedAt: oldestUpdatedAt },
    };
```